simulator/hart_impl.cpp

```diff
@@ -20,11 +20,19 @@ void Hart::RunImpl(Mode mode)
         case Mode::BB: {
             interpreter::BB raw_bb;
             interpreter::DecodedBB decodedBB;
+            Register cache_addr;
             do {
-                fetch_.loadBB(executor_.getPC(), raw_bb);
-                decoder_.DecodeBB(raw_bb, decodedBB);
-                executor_.RunBB(decodedBB);
-                counter += raw_bb.size();
+                cache_addr = executor_.getPC() / 4 % BB_CACHE_SIZE;
+                if (auto &&el = bb_cache_[cache_addr]; el.first == executor_.getPC()) {
```

Comment:
I think that this change will make it more readable
```suggestion
                if (auto &&[addr, decoded_bb]= bb_cache_[cache_addr]; addr == executor_.getPC()) {
```

Reply:
And 2 lines bottom offcourse the same